src/backend/distributed/connection/remote_commands.c

```diff
@@ -75,6 +75,90 @@ ForgetResults(MultiConnection *connection)
 
 
 /*
+ * NonblockingForgetResults clears a connection from pending activity if doing
+ * so does not require network IO. Returns true if successful, false
+ * otherwise.
+ */
+bool
+NonblockingForgetResults(MultiConnection *connection)
+{
+	PGconn *pgConn = connection->pgConn;
+	bool wasNonblocking = PQisnonblocking(pgConn);
+
+	if (PQstatus(pgConn) != CONNECTION_OK)
+	{
+		return false;
+	}
+
+	while (true)
+	{
+		PGresult *result = NULL;
+
+		CHECK_FOR_INTERRUPTS();
```

Comment:
Given that this loop doesn't block, do we need `CHECK_FOR_INTERRUPTS` inside the loop? It would simplify the code if we can't error out here (e.g. we can move `PQsetnonblocking` outside the loop).